include/itkAttenuationImageFilter.hxx

```diff
@@ -0,0 +1,312 @@
+{
+  return this->TransformPixelToPhysicalScanLineDistance(this->GetPadUpperBounds());
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::SetPadLowerBoundsMM(const float distanceMM)
+{
+  this->SetPadLowerBounds(TransformPhysicalToPixelScanLineDistance(distanceMM));
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+float
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::GetPadLowerBoundsMM() const
+{
+  return this->TransformPixelToPhysicalScanLineDistance(this->GetPadLowerBounds());
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+const ImageRegionSplitterBase *
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::GetImageRegionSplitter() const
+{
+  m_RegionSplitter->SetDirection(m_Direction);
+  return m_RegionSplitter.GetPointer();
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::BeforeThreadedGenerateData()
+{
+  // Verify inputs
+  if (this->GetMaskImage() == nullptr)
+  {
+    itkExceptionMacro("Filter requires a mask image for inclusion estimates!");
+  }
+  else
+  {
+    m_ThreadedMaskImage = this->GetMaskImage();
+  }
+
+  if (this->GetDirection() >= ImageDimension)
+  {
+    itkExceptionMacro("Scan line direction must be a valid image dimension!");
+  }
+
+  if (this->GetSamplingFrequencyMHz() < itk::Math::eps)
+  {
+    itkExceptionMacro("RF sampling frequency was not set!");
+  }
+
+  Superclass::BeforeThreadedGenerateData();
+
+  // Initialize metric image
+  this->GetOutput()->Allocate();
+  this->GetOutput()->FillBuffer(0.0f);
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::DynamicThreadedGenerateData(
+  const OutputRegionType & regionForThread)
+{
+  if (regionForThread.GetNumberOfPixels() == 0)
+  {
+    return;
+  }
+
+  const MaskImageType *  maskImage = this->GetMaskImage();
+  const InputImageType * input = this->GetInput();
+  OutputImageType *      output = this->GetOutput();
+
+  ImageLinearConstIteratorWithIndex<TInputImage> it(input, regionForThread);
+  it.SetDirection(m_Direction);
+  it.GoToBegin();
+
+  unsigned int   inclusionLength;
+  InputIndexType start, end;
+
+  const float scanStepMM = input->GetSpacing()[m_Direction];
+
+  // do the work
+  while (!it.IsAtEnd())
+  {
+    inclusionLength = 0;
+    while (!it.IsAtEndOfLine())
+    {
+      // Advance until an inclusion is found
+      InputIndexType index = it.GetIndex();
+      if (ThreadedIsIncluded(index) && inclusionLength == 0)
+      {
+        // Step into inclusion
+        start = it.GetIndex();
+      }
+      else if (!ThreadedIsIncluded(index) && inclusionLength > 0)
+      {
+        // Step out of inclusion
+        end = it.GetIndex();
```

Comment:
Reviewing once more, I think `end` needs to move backwards by one pixel so that we are only using pixels inside the mask for estimation.